yt_dlp/extractor/nekohacker.py

```diff
@@ -0,0 +1,125 @@
+from datetime import datetime
+import re
+
+from yt_dlp.utils import ExtractorError, determine_ext, extract_attributes, get_element_by_class, unified_strdate
+from .common import InfoExtractor
+
+
+class NekoHackerIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?nekohacker\.com/(?P<id>[a-zA-Z0-9-_]+)'
+    _ENTRY_URL = r'https?://(?:www\.)?nekohacker\.com/wp-content/uploads/\d+/\d+/(?P<id>(?:\d+-)?[a-zA-Z0-9_-]+)\.[a-zA-Z0-9]+'
```

Comment:
```suggestion
```
I think this is the wrong way of going about extracting an entry `id` value for a few reasons:
 - This is not necessarily going to be unique, e.g. the first test's id is `01-Spaceship`, but what if another release also has a first track named "Spaceship"?
 - The media URLs won't always match this pattern (see https://nekohacker.com/susume/)
 - The `<li>` elements have a `data-trackid` attribute that does seem to be a unique identifier for this site, and IMO we should use that instead